integration-tests/utils/helpers.ts

```diff
@@ -0,0 +1,173 @@
+import { getEnvVariable } from '../lib/zendeskParameters'
+import { initiateDataRequestLambdalogGroupName } from '../lib/cloudWatchParameters'
+import { cloudWatchLogsClient } from './cloudWatchLogsClient'
+import {
+  DescribeLogStreamsCommandInput,
+  DescribeLogStreamsCommandOutput,
+  DescribeLogStreamsCommand,
+  LogStream,
+  FilterLogEventsCommandInput,
+  FilterLogEventsCommand,
+  FilterLogEventsCommandOutput,
+  FilteredLogEvent
+} from '@aws-sdk/client-cloudwatch-logs'
+
+const generateRandomNumber = () => {
+  return Math.floor(Math.random() * 100).toString()
+}
+
+const authoriseAs = (username: string) => {
+  return Buffer.from(
+    `${username}/token:${getEnvVariable('ZENDESK_TEST_API_TOKEN')}`
+  ).toString('base64')
+}
+
+const generateZendeskRequestDate = (offset: number): string => {
+  const today: Date = new Date()
+  today.setDate(today.getDate() + offset)
+
+  const dateFormat: Intl.DateTimeFormat = new Intl.DateTimeFormat('en-GB')
+  const dateParts: string[] = dateFormat.format(today).split('/')
+  return `${dateParts[2]}-${dateParts[1]}-${dateParts[0]}`
+}
+
+const getLogStreamPrefix = () => {
+  const dateFormat = new Intl.DateTimeFormat('en-GB')
+  const dateParts = dateFormat.format(new Date()).split('/')
+  return `${
+    dateParts[2] + '/' + dateParts[1] + '/' + dateParts[0] + '/' + '[$LATEST]'
+  }`
+}
+
+const getLatestLogStreamName = async (): Promise<string> => {
+  const describeLogStreamsParams: DescribeLogStreamsCommandInput = {
+    logGroupName: initiateDataRequestLambdalogGroupName,
+    orderBy: 'LastEventTime',
+    descending: true,
+    limit: 1
+  }
+
+  const describeLogStreamsCommand = new DescribeLogStreamsCommand(
+    describeLogStreamsParams
+  )
+
+  pause(500)
+  const describeLogStreamsResponse: DescribeLogStreamsCommandOutput =
+    await cloudWatchLogsClient.send(describeLogStreamsCommand)
+
+  const latestLogStreams: LogStream[] =
+    describeLogStreamsResponse.logStreams ?? []
+
+  expect(latestLogStreams.length).toEqual(1)
+  return latestLogStreams[0].logStreamName as string
+}
+
+interface EventLogStream {
+  logStreamName: string
+  eventMessage: string
+}
+
+const waitForLogStreamContainingEvent = async (
+  eventFilterPattern: string,
+  ...eventMessagePatterns: string[]
+): Promise<EventLogStream> => {
+  if (eventMessagePatterns.length == 0) {
+    throw Error('No message patterns provided for event')
+  }
+
+  let latestLogStreamName = await getLatestLogStreamName()
+  console.log(`LATEST LOG STREAM NAME: ${latestLogStreamName}`)
+  let eventMatched = false
+  // eslint-disable-next-line @typescript-eslint/no-inferrable-types
+  let eventMessage: string = ''
```

Comment:
You can drop the type and remove the linter disable comments on these as typescript will infer the type and then you can't easily change it.

Reply:
I had done that but the code wouldn't compile due to `possibly unassigned` error.  I had to assign it to avoid the error. I couldn't think of any other way to clear the error.